### DISCUSSION TITLE
Long test.py and missing text highlights

### DISCUSSION THREAD

**Discussion by Niferu** [smartcontractkit/full-blockchain-solidity-course-py#1654] [Q&A] [answered]
Hello,

I have 2 silly problems that I would like to solve as I'm perfectionist and it drives me crazy.

**1. In course Patrick runs multiple test.py files and it takes around 3s to finish. In my case those tests are 3 times longer, my pc is pretty good and i couldn't find what can be the problem here.**

Patrick "test_get_entrance_fee" Test:
![image](https://user-images.githubusercontent.com/110537615/185259833-833516b3-3ac8-4655-af33-bafe10625b3f.png)

My "test_get_entrance_fee" Test:
![image](https://user-images.githubusercontent.com/110537615/185259875-7fc77e59-7392-448e-ab8b-11b78703e8fc.png)

**2. I'm also having problem with text formatting as below, some key words are missing from my theme or smth. Any idea how to setup it correctly to get those key words highlighted as it is in course video?**

Formatting from video:
![image](https://user-images.githubusercontent.com/110537615/185259626-5e03fb4c-d79a-4831-954c-204944f70e53.png)

My formatting: (I'm using Dark+ theme)
![image](https://user-images.githubusercontent.com/110537615/185259721-b7eb0425-eed7-4a08-99ff-bc0db59f572e.png)

**Comment by cromewar** [comment]
Hello @Neftyr 
Well those are tricky questions, but I'll try to set  point of view into them.
1. the time it take the tests to run depends on a lot of factors, not just the specs of the pc. The operating systems, the amount of subprocess you have running, the architecture of your OS and more...

For example I have a intel 11 gen processor on my main tower, is a 6 core 12 threads which may fall into premium mid range, and I also have a M1 iMac 2021. When I run the tests my iMac usually is more faster than the main tower, even when in paper the brute force on my main tower is better than the iMac.

Why? Well again it depends on a lot of factors, 
On my main tower I run the code using WSL (windows subsystem for linux) which inevitably will put more processor stress into the machine, also windows has a lot of background process running and so on. Also I tested the same code on the main tower using native Linux (Arch Linux) and it ran faster than on windows with WSL.

So, if you are a performance snob I would recommend you to look into more factors rather than the brute force the machine has in paper. I would highly recommend to give a look into M1 pro Macs if you like apple, or Lightway Linux distros if you are on the Pc ecosystem.
Just take in mind, that in the end... this does not matter that much.

2. You can personalize the theme you are using on vscode to look how ever you like, you must open `settings.json` and add a rule for the theme you are using, you need to get the scope of each element you want to modify and the modifications will look something like this:

![image](https://user-images.githubusercontent.com/20199180/185439667-0bb58871-e9b8-4231-abfe-f0dae317c21a.png)

for semantic modifications, like this:

![image](https://user-images.githubusercontent.com/20199180/185439828-166ea07c-8687-4117-94de-27fd2e0a9dd6.png)

Of course you will have to know the rules and the scopes you want to modify, this video might help: https://www.youtube.com/watch?v=mC_htrJ1QPg

Cheers, and I hope this info might help.

**Reply by Niferu** [reply]
Hi @cromewar,

First of all thanks a lot for looking into it, much appreciate it!

Regarding 1st problem, amount of opened processes doesn't matter, I have ran testing under huge hardware usage and then turned off everything in order to free as much resources as I could and results was the same (still around 11s). I'm using windows, so that might be the case here as Patrick is working on Mac i believe.

Regarding 2nd problem I was following course fully and I didn't use VS Code previously, so I thought Patrick may have some kind of extensions maybe that I'm missing. I have checked VS Code setup and settings with my friend who is going thru course with me and we are setup the same, but for him syntax is correct and matches one from video. We have checked syntax info following video you have sent and below are the results:

My syntax info:
![image](https://user-images.githubusercontent.com/110537615/185470189-a44feeff-a785-4ccd-a316-47d4c19bf6ed.png)

My friend syntax info:
![image](https://user-images.githubusercontent.com/110537615/185470339-af7e4663-ecc6-4ada-bf47-f96e46c880a7.png)

If you have any idea where is it come from and what i have to change let me know. Our json files looks the same no custom syntax added. Last thing is interpreter as I'm using global and my friend is using recommended. 
![image](https://user-images.githubusercontent.com/110537615/185470607-4986cf25-fdba-4107-9174-baa71443400e.png)

The case for me here is that if I will switch to recommended for example "brownie" and "pytest" are like below. So maybe that is the case? As I'm newbie I cannot tell, so appreciate your help.
![image](https://user-images.githubusercontent.com/110537615/185471054-753fa6b5-6011-4ad5-adb6-e67f6fc8644b.png)

**Reply by Niferu** [reply]
Ok i have figured out it is for sure interpreter, but i have no idea how to fix pathing for recommended one..

**Comment by Niferu** [comment, marked as answer]
Hello Everyone!

I have managed to solve all the issues.

1. First problem as I already said above and thanks to @cromewar is simply operating system related and tests runs faster on Mac than on Windows.

2. Second problem, I was installing brownie after downloading VS Studio tools with "pipx install brownie", previously i was trying to install brownie by commands like "pip install brownie" or "pip3 install eth-brownie", but it didn't work as i didn't have VS Studio tools installed yet. Simply the case here was changing interpreter manually for "~/.local/pipx/venvs/eth-brownie/bin/python", but i went further as i didn't want to add this interpreter to every single project. The fix for this was simply installing brownie with "pip3 install eth-brownie" and now it worked because i had VS Studio tools installed now and as it was installed by pip3 pathing was correct now for my manually installed python. I have also removed windows store versions of python, and now I'm using below only.

![image](https://user-images.githubusercontent.com/110537615/185655920-c31d394a-efb6-4c6a-95ad-46eecc9ca133.png)

I hope this will help someone!
